### DISCUSSION TITLE
Delay given for event scheduling

### DISCUSSION THREAD

**Discussion by MaxMart99** [Infineon/optiga-trust-m#99] [Q&A] [answered]
Hello,

I have been working with an Optiga Trust M for 2 months now. I think I start to have quite a good understanding of how Infineon's software works, but since I started I had a question about the delay system used for the callback of events.
Indeed, the function used to schedule future events has the following prototype:

```
/**
 * \brief Callback registration function to trigger once when timer expires.
 *
 * \details
 * Platform specific event call back registration function to trigger once when timer expires.
 *  - This function registers the callback function supplied by the caller.
 *  - It triggers a timer with the supplied time interval in microseconds.
 *  - Once the timer expires, the registered callback function gets called.
 *
 * \pre
 * - None
 *
 * \note
 * - None
 *
 * \param[in] p_pal_os_event        Pointer to pal_os_event
 * \param[in] callback              Callback function pointer
 * \param[in] callback_args         Callback arguments
 * \param[in] time_us               time in micro seconds to trigger the call back
 *
 */
LIBRARY_EXPORTS void pal_os_event_register_callback_oneshot(pal_os_event_t * p_pal_os_event,
                                                            register_callback callback,
                                                            void * callback_args,
                                                            uint32_t time_us);
```

According to this, the time given in parameters are in microseconds. However, when reading the different delays given in ```optiga-trust-m/optiga/include/optiga/ifx_i2c/ifx_i2c_config.h```, you can find some values in microseconds, and others in milliseconds.

```
/** @brief Physical Layer: polling interval in microseconds */
#define PL_POLLING_INVERVAL_US      (1000U)
/** @brief Physical Layer: data register polling interval in microseconds */
#define PL_DATA_POLLING_INVERVAL_US (5000U)
/** @brief Physical Layer: guard time interval in microseconds */
#define PL_GUARD_TIME_INTERVAL_US   (50U)
/** @brief Data link layer: Trans timeout in milliseconds*/
#define PL_TRANS_TIMEOUT_MS         (10U)
/** @brief Start up time */
#define STARTUP_TIME_MSEC           (12000U)
```

STARTUP_TIME_MSEC is in milliseconds and PL_POLLING_INVERVAL_US is in microseconds, but they are both used in the same ```pal_os_event_register_callback_oneshot``` function, which is supposed to take time as microseconds:
```
pal_os_event_register_callback_oneshot( p_ctx->pal_os_event_ctx,
                                                    ifx_i2c_pl_negotiation_event_handler,
                                                   ((void * )p_ctx),
                                                   PL_POLLING_INVERVAL_US);
```
```
pal_os_event_register_callback_oneshot(p_ctx->pal_os_event_ctx,(register_callback)ifx_i2c_pl_negotiation_event_handler,
                                                           (void * )p_ctx,
                                                           STARTUP_TIME_MSEC);
```
Hence, I don't really now how to understand it. Does the STARTUP_TIME_MSEC really takes 12 seconds, is it just a mistake in the name of the variable, or am I understanding something wrong  ?

Thanks,
Maxime

**Comment by ShaikKarishma0** [comment, marked as answer]
Hi @MaxMart99,

All the time values that are passed to pal_os_event_register_callback_oneshot() function are in microseconds, as the function definition mentions.

The startup time value is also in microseconds and STARTUP_TIME_MSEC is a typo. It shall be renamed to STARTUP_TIME_USEC.

**Reply by MaxMart99** [reply]
Alright, thank you very much for your answer, it makes more sense to me now !

Regards,
Maxime
